recipes/openslide/all/test_package/test_package.c

```diff
@@ -0,0 +1,6 @@
+#include <openslide.h>
+
+int main() {
+    openslide_detect_vendor("");
```

Comment:
This doesn't really test anything.  You'd need to do `openslide_open("")` instead, with the `OPENSLIDE_DEBUG` environment variable set to `synthetic`, and you'd need to check whether the open succeeded.  `meson test` will do all of this for you, and I'd recommend using that rather than hand-rolling your own test.  If you want to avoid running tests at build time, you could run `OPENSLIDE_DEBUG=synthetic slidetool prop list ""` instead.

Reply:
> This doesn't really test anything.

As intended. :smile: `test_package` exists only to confirm that compilation, linking and dynamic linking during execution actually work.

As for unit tests - CCI unfortunately does not allow running actual tests as part of the build since it would be too taxing for the CI given the quite large build matrix.

Reply:
Ah, got it.  If the rules don't absolutely forbid it, you might still consider running the built-in smoke test though.  Its purpose is exactly to check that each of the dependencies is linked correctly and can read a built-in 16x16 test image.  `meson test` runs _only_ that test right now (most of our tests require a lot of slide data), but if you want to be future-proof, you could avoid `meson test` and directly open the zero-length filename as mentioned above.  On my machine, invoking it via `slidetool prop list ""` takes less than 30 ms of CPU time.